llpc/lower/llpcSpirvProcessGpuRtLibrary.cpp

```diff
@@ -349,6 +351,85 @@ void SpirvProcessGpuRtLibrary::createConvertF32toF16WithRoundingMode(Function *f
   m_builder->CreateRet(result);
 }
 
+// =====================================================================================================================
+// Create function to do the float op with the round Mode.
+//
+// @param func : The function to process
+void SpirvProcessGpuRtLibrary::createFloatOpWithRoundMode(Function *func) {
+  // void AmdExtD3DShaderIntrinsics_FloatOpWithRoundMode(uint roundMode, uint operation, float src0, float src1))
+  // {
+  //     switch (roundMode)
+  //     {
+  //         case AmdExtD3DShaderIntrinsicsFloatOpWithRoundMode_TiesToEven:
+  //             dest.SetRoundMode(RTE);
+  //             switch (operation)
+  //             {
+  //                 case AmdExtD3DShaderIntrinsicsFloatOpWithRoundMode_Add:
+  //                     dest = src0 + src1
+  //                     break;
+  //                 case AmdExtD3DShaderIntrinsicsFloatOpWithRoundMode_Subtract:
+  //                     dest = src0 - src1
+  //                     break;
+  //                 ...
+  //             }
+  //             break;
+  //         case AmdExtD3DShaderIntrinsicsFloatOpWithRoundMode_TowardPositive:
+  //             dest.SetRoundMode(RTP);
+  //             ...
+  //     }
+  // }
+  enum OperationType : uint32_t { Add = 0, Sub, Mul };
+  auto int32Ty = m_builder->getInt32Ty();
+  auto retType = cast<FixedVectorType>(func->getReturnType());
+  auto argIt = func->arg_begin();
+  Value *roundMode = m_builder->CreateLoad(int32Ty, argIt++);
+  Value *operation = m_builder->CreateLoad(int32Ty, argIt++);
+  Value *src0 = m_builder->CreateLoad(retType, argIt++);
+  Value *src1 = m_builder->CreateLoad(retType, argIt);
+
+  Value *fpExcpt = MetadataAsValue::get(*m_context, MDString::get(*m_context, "fpexcept.strict"));
```

Comment:
Do we really have shader code that reads the floating point exception flags?

Reply:
I think there is no excption flags for shader

Reply:
The shader HW does have exception flags. But if we don't read them, this should be "fpexcept.ignore" instead of "fpexcept.strict" to allow for a little more freedom in codegen.

Reply:
another question is constrained.fp seems fail to generate SP3 to set roundmode.